File: TUF A2Z DSA Sheet/basics/recursion problems/reverse_array.cpp
// Program: Reverse the given array.

#include <bits/stdc++.h>
using namespace std;

int reverse_array(int arr[], int start, int end) {
    if(start < end) {
        swap(arr[start], arr[end]);
        reverse_array(arr, start+1, end-1);
    }
}

int main() {
    
    int n = 6;
    int arr[] = {33, 44, 21, 54, 11, 43};
    reverse_array(arr, 0, n-1);
    for(int i=0; i<n; i++){
        cout << arr[i] << " ";
    }
    return 0;
}

File: DSA Implementation in Go/Arrays/reverse_array.go
//go:build ignore

package main

import "fmt"

func reverseArray(arr []int) []int {
	start := 0
	end := len(arr) - 1
	for start < end {
		arr[start], arr[end] = arr[end], arr[start]
		start++
		end--
	}
	return arr
}

func main() {
	arr := []int{1, 2, 3, 4, 5}
	
	reversed := reverseArray(arr)
	fmt.Printf("Reversed array: %v\n", reversed)
}
